docs/index.rst

```diff
@@ -3,8 +3,128 @@
+use the statement ``import unumpy as np`` instead of the usual
+``import numpy as np``. This is deliberate: it makes dispatching opt-in
+instead of being forced to use it, and the overhead associated with it.
+However, a package is free to define its main methods as the dispatchable
+versions, thereby allowing dispatch on the default implementation.
+
+Extensibility *and* Choice
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If some effort is put into the dispatch machinery, it's possible to
+dispatch over arbitrary objects --- including arrays, dtypes, and
+so on. A method defines the type of each dispatchable argument, and
+backends are *only* passed types they know how to dispatch over.
+
+Methods can have a default implementation in terms of other methods,
+but they're still overridable.
+
+This means that only one framework is needed to, for example, dispatch
+over ``ufunc`` s, arrays, dtypes and all other primitive objects in NumPy,
+while keeping the core ``uarray`` code independent of NumPy and even ``unumpy``.
+
+Backends can span modules, so SciPy could jump in and define its own methods
+on NumPy objects and make them overridable within the NumPy backend.
+
+User Choice
+^^^^^^^^^^^
+
+The users of ``unumpy`` or ``uarray`` can choose which backend they want
+to prefer with a simple context manager. They also have the ability to
+force a backend, and to skip a backend. This is useful for array-like
+objects that provide other array-like objects by composing them. For
+example, Dask could perform all its blockwise function calls with the
+following psuedocode (obviously, this is simplified):
+
+.. code:: python
+
+    in_arrays = extract_inner_arrays(input_arrays)
+    out_arrays = []
+    for input_arrays_single in in_arrays:
+        args, kwargs = blockwise_function.replace_args_kwargs(
+            args, kwargs, input_arrays_single)
+        with ua.skip_backend(DaskBackend):
+            out_arrays_single = blockwise_function(*args, **kwargs)
+        out_arrays.append(out_arrays_single)
+
+    return combine_arrays(out_arrays)
+
+A user would simply do the following:
+
+.. code:: python
+
+    x = da.array([a, b, c])
+
+    # Anything that uses x will (usually) use the Dask backend
+
+    with ua.use_backend(DaskBackend):
+        # Write all your code here
+        # It will prefer the Dask backend
+
+Addressing past flaws
+^^^^^^^^^^^^^^^^^^^^^
+
+The progress on NumPy's side for defining an override mechanism has been slow, with
+NEP-13 being first introduced in 2013, and with the wealth of dispatchable objects
+(including arrays, ufuns and dtypes), and with the advent of libraries like Dask,
+CuPy, XArray, PyData/Sparse and Xnd, it has become clear that the need for alternative
+array-like implementations is growing. There are even other libraries like PyTorch, and
+TensorFlow that'd be possible to express in NumPy API-like terms. Another example includes
+the Keras API.
```

Comment:
The PyTorch and TensorFlow ones I get, although it should say "mostly". Keras is very different, so I would not use that here (functionality like `model.add/compile/evaluate/predict`, its generator classes, etc. don't have a NumPy equivalent).

Reply:
What I mean is you could have a `ukeras` that depended on `unumpy`/`uscipy` which was also overridable. 🙂